herd_behaviour.nlogo

```diff
@@ -27,37 +29,87 @@ to setup
     set v-dir pseudo-random-vector
     set color blue
     set w 0
+    set g-dir [0 0]
   ]
   let number_informed_members (number_herd_members * ( 0.01 * (percent_informed)))
   ask n-of number_informed_members turtles [
     set w  weight
     set color red
+    set g-dir [125 125]
   ]
 end
 
 to go
   ask turtles[
-    ifelse count other turtles in-radius avoidance_range > 0 [
+    let base [0 0] ;; base of function calculated in every
+    let ci-dir array:from-list (list xcor ycor) ;; the position vector of actual turtle
+    ifelse any? other turtles in-radius avoidance_range [
       ;; avoid them
-      calculate-dir
+      ask turtles in-radius avoidance_range [
+        let cj-dir array:from-list (list xcor ycor)
+        set base plus-vectors base calculate-base ci-dir cj-dir;; add the dividing by absolute vector
+      ]
+      set temp minus-one base
+      set d-dir unit-vector temp
+      ;; for informed turtles, extra formula 3
+      if w != 0[
```

Comment:
Ich hätte es so verstanden, dass sich die erste Formel ausschließlich in Abhängigkeit der auszuweichenden Nachbarn berechnet, also ohne bevorzugte Richtung.